packages/functional-tests/pages/signinTokenCode.ts

```diff
@@ -2,51 +2,13 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-import { BaseLayout } from './layout';
+import { BaseTokenCodePage } from './baseTokenCode';
 
-export class SigninTokenCodePage extends BaseLayout {
-  readonly path = 'sign_in_token_code';
-
-  readonly selectors = {
-    LINK_RESEND: '#resend',
-    SIGNIN_HEADER: '#fxa-signin-header',
-    SIGNIN_TOKEN_CODE_HEADER: '#fxa-signin-code-header',
-    SIGNIN_PASSWORD_HEADER: '#fxa-signin-password-header',
-    SUBMIT: 'button[type=submit]',
-    TEXT_INPUT: 'input[type=text]',
-    TOOLTIP: '.tooltip',
-    SUCCESS: '.success',
-  };
-
-  get passwordHeader() {
-    return this.page.locator(this.selectors.SIGNIN_PASSWORD_HEADER);
-  }
-
-  get tokenCodeHeader() {
-    return this.page.locator(this.selectors.SIGNIN_TOKEN_CODE_HEADER);
-  }
-
-  get input() {
-    return this.page.locator(this.selectors.TEXT_INPUT);
-  }
-
-  get submit() {
-    return this.page.locator(this.selectors.SUBMIT);
-  }
-
-  get tooltip() {
-    return this.page.locator(this.selectors.TOOLTIP);
-  }
+export class SigninTokenCodePage extends BaseTokenCodePage {
+  readonly path = '/signin_token_code';
 
   get resendLink() {
-    return this.page.locator(this.selectors.LINK_RESEND);
-  }
-
-  get successMessage() {
-    return this.page.locator(this.selectors.SUCCESS);
-  }
-
-  async clickSubmitButton() {
-    await this.page.locator(this.selectors.SUBMIT).click();
+    this.checkForPath(this.path);
+    return this.page.locator('#resend');
```

Comment:
Here we extend the base class, since this action was speficially need when testing the the signin token page.